manual_tag_info_gather/manual_tag_info_collector.py

```diff
@@ -0,0 +1,75 @@
+"""
+For all users assigned tagging work, check how many tags are left, and how many require pruning.
+Also, generate an aggregate ecosystem_package_topic_map for PGM training.
+"""
+"""
+To run the script:
+Add the config parameters to env
+cp config.py.template config.py
+All the user_name as user_name = ["name1", "name2",...]
+python manual_tag_info_collector.py
+"""
+
+import config
+import requests
+import json
+import sys
+
+# Update variables from config.
+user_names = config.user_names
+prefix_url = config.prefix_url
+ecosystem = config.ecosystem
+Authorization = config.Authorization
+
+header = {'Authorization': Authorization}
+user_data_not_found = []
+final_data_list = []
+for_each_ecosystem = {'ecosystem': ecosystem}
+final_package_dict = {}
+
+# For each user, collect the package_topic_map list
+try:
+    for name in user_names:
+        url = prefix_url + name + "/" + ecosystem
+        response = requests.get(url, headers=header)
+        if response.status_code != 200:
+            user_data_not_found.append(name)
+            continue
+        response_data = json.loads(response.content)
+        package_topic_map = response_data.get(
+            'data', {}).get('package_topic_map', {})
+        for package_name, tag_list in package_topic_map.items():
+            if package_name in final_package_dict.keys():
+                tag_list = tag_list.extend(final_package_dict[package_name])
+            final_package_dict[package_name] = list(set(tag_list))
+except Exception as e:
+    raise e
```

Comment:
What's the reason for catching and raising any exception ?

Reply:
In case the endpoint is not reachable. Faced this issue in the morning, for some time, the connection failed.

Reply:
OK, I understand you want to catch any unexpected problems.
But you simply raise the catched exception again, so it behaves exactly the same as without any `try: except:`.
I mean:
This code
```python
1/0
```
behaves exactly the same as this code:
```python
try:
    1/0
except Exception as e:
    raise e
```
i.e. in both cases you get `ZeroDivisionError` exception.

So all I want to say is that you can remove the `try: except: raise:` and still have the same behaviour.

If you want to just silence the exception (which I'd advise against), you need something like `pass` or `print("unexpected exception")` instead of the `raise`.

Reply:
OK, I understand you want to catch any unexpected problems.
But you simply raise the catched exception again, so it behaves exactly the same as without any `try: except:`.
I mean:
This code

```python
1/0
```
behaves exactly the same as this code:

```python
try:
    1/0
except Exception as e:
    raise e
```

i.e. in both cases you get `ZeroDivisionError` exception.

So all I want to say is that you can remove the `try: except: raise:` and still have the same behaviour.

If you want to just silence the exception (which I'd advise against), you need something like `pass` or `print("unexpected exception")` instead of the `raise`.